database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala

```diff
@@ -173,30 +173,144 @@ class MetadataSlickDatabase(originalDatabaseConfig: Config)
+                                   labelMetadataKey: String,
+                                   limit: Int,
+                                   buildUpdatedSummary:
+                                   (Option[WorkflowMetadataSummaryEntry], Seq[MetadataEntry])
+                                     => WorkflowMetadataSummaryEntry)
+                                  (implicit ec: ExecutionContext): Future[Long] = {
     val action = for {
-      previousMetadataEntryIdOption <- getSummaryStatusEntryMaximumId(
-        "WORKFLOW_METADATA_SUMMARY_ENTRY", "METADATA_ENTRY")
-      previousMetadataEntryId = previousMetadataEntryIdOption.getOrElse(-1L)
-      metadataEntries <- dataAccess.metadataEntriesForIdGreaterThanOrEqual((
-        previousMetadataEntryId + 1L, startMetadataKey, endMetadataKey, nameMetadataKey, statusMetadataKey, likeMetadataLabelKey, submissionMetadataKey)).result
-      metadataWithoutLabels = metadataEntries.filterNot(_.metadataKey.contains(labelMetadataKey)).groupBy(_.workflowExecutionUuid)
-      customLabelEntries = metadataEntries.filter(_.metadataKey.contains(labelMetadataKey))
-      _ <- DBIO.sequence(metadataWithoutLabels map updateWorkflowMetadataSummaryEntry(buildUpdatedSummary))
-      _ <- DBIO.sequence(customLabelEntries map toCustomLabelEntry map upsertCustomLabelEntry)
-      maximumMetadataEntryId = previousOrMaximum(previousMetadataEntryId, metadataEntries.map(_.metadataEntryId.get))
-      _ <- upsertSummaryStatusEntryMaximumId(
-        "WORKFLOW_METADATA_SUMMARY_ENTRY", "METADATA_ENTRY", maximumMetadataEntryId)
+      previousMetadataEntryIdOption <- getSummaryStatusEntrySummaryPosition(summarizeNameIncreasing)
+      previousMaxMetadataEntryId = previousMetadataEntryIdOption.getOrElse(-1L)
+      nextMaxMetadataEntryId = previousMaxMetadataEntryId + limit
+      maximumMetadataEntryId <- summarizeMetadata(
+        minMetadataEntryId = previousMaxMetadataEntryId + 1L,
+        maxMetadataEntryId = nextMaxMetadataEntryId,
+        startMetadataKey = startMetadataKey,
+        endMetadataKey = endMetadataKey,
+        nameMetadataKey = nameMetadataKey,
+        statusMetadataKey = statusMetadataKey,
+        submissionMetadataKey = submissionMetadataKey,
+        parentWorkflowIdKey = parentWorkflowIdKey,
+        rootWorkflowIdKey = rootWorkflowIdKey,
+        labelMetadataKey = labelMetadataKey,
+        buildUpdatedSummary = buildUpdatedSummary,
+        summaryPositionFunction =
+          metadataEntries => {
+            if (metadataEntries.nonEmpty) {
+              DBIO.successful(metadataEntries.map(_.metadataEntryId.get).max)
+            } else {
+              /*
+              Nothing seen within the window of (previousMetadataEntryId, nextMaxMetadataEntryId].
+              Check if there are more entries above nextMaxMetadataEntryId.
+              If yes, then start next time after nextMaxMetadataEntryId, otherwise reuse previousMetadataEntryId.
+               */
```

Comment:
Ran into it.

Some of the call-caching metadata dumps end up containing more rows than a "window". Because the summarizer kept looking at the same start point, it ended up retrying the same window with no metadata-rows-to-be-summarized, wedging _multiple_ centaurs: https://travis-ci.com/broadinstitute/cromwell/builds/104886244